app/battles.py

```diff
@@ -0,0 +1,24 @@
+from app.knight import Knight
+
+
+def is_healthy(knight: Knight) -> int:
+    if knight.hp <= 0:
+        knight.hp = 0
+    return knight.hp
+
+
+def one_pair_battle(knight1: Knight, knight2: Knight) -> dict:
+    knight1.hp -= knight2.power - knight1.protection
+    knight2.hp -= knight1.power - knight2.protection

```

Comment:
Ensure that the calculation of `hp` does not result in negative values. Consider using `max(0, calculated_hp)` to prevent `hp` from going below zero.